test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/NestedTypesTest.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Sdk.Generators;
+using Xunit;
+
+namespace Microsoft.Azure.Functions.SdkGeneratorTests
+{
+    public partial class FunctionMetadataProviderGeneratorTests
+    {
+        public class NestedTypesTests
+        {
+            private readonly Assembly[] _referencedExtensionAssemblies;
+
+            public NestedTypesTests()
+            {
+                // load all extensions used in tests (match extensions tested on E2E app? Or include ALL extensions?)
```

Comment:
whatever extensions are used in the inputCode snippets in the tests in this file need to be loaded and others can be dropped.

Reply:
I think we need the 6 we are loading. Do you notice anything which is not needed here?

Reply:
I was responding to the comment - is that copy/pasted from somewhere? It reads like my writing ... so if it is copy/pasted will note that comment for a cleanup PR